src/reanimated2/core.js

```diff
@@ -150,3 +151,19 @@ export function startMapper(mapper, inputs = [], outputs = []) {
 export function stopMapper(mapperId) {
   NativeReanimated.stopMapper(mapperId);
 }
+
+export let runOnJS = (fun) => fun;
+
+if (Platform.OS !== 'web') {
+  runOnJS = (fun) => {
+    'worklet';
+    if (!_WORKLET) { // eslint-disable-line
+      return fun;
+    }
+    if (!fun.__callAsync) {
+      throw new Error("Function couldn't be called by runOnJS");
```

Comment:
Would prefer for error to be more descriptive than this. In this case we throw here because someone tried to us `runOnJS` on something that is not a host function. Something along these lines would give people better pointers:
"Attempting to call runOnJS with an object that is not a host function. Using runOnJS is only possible with methods that are defined on the main React-Native Javascript thread and that aren't marked as worklets"